src/taskgraph/config.py

```diff
@@ -40,6 +40,11 @@
             description="Default 'deadline' for tasks, in relative date format. "
             "Eg: '1 week'",
         ): optionally_keyed_by("project", str),
+        Optional(
+            "task-expires-after",
+            description="Default 'expires-after' for non level 3 tasks, in relative date format. "
+            "Eg: '90 days'",
+        ): str,
```

Comment:
I think saying "non level 3" is technically incorrect because `is_try()` only catches pull requests and projects that contain the string "try". Saying "level 1 tasks" would be more accurate.

That said, it's kind of a shame that this only impacts level 1 tasks. What if we added an `optionally_keyed_by("level", str)` (like how `task-deadline-after` is optionally keyed by project above).

This way the graph config could look like:
```
task-expires-after:
    by-level:
        1: 90 days
        3: 1 year
```

Also, you could just set a single default for all levels:
```
task-expires-after: 1 year
```

Feel free to leave as a follow-up, but would be nice to have an issue on file at least.

Reply:
I looked into it a bit and filed #410